src/ERC1271.sol

```diff
@@ -1,50 +1,77 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.4;
 
-/// @notice ERC-1271 with guards for same signer being used on multiple accounts
-/// Based on Solady (https://github.com/vectorized/solady/blob/main/src/accounts/ERC1271.sol)
+/// @title ERC-1271 With Cross Account Replay Protection
+///
+/// @notice Abstract ERC-1271 implementation (based on Solady's one) with guards to handle the same
+///         signer being used on multiple accounts, using a nested approach based on EIP-712.
+///
+/// @dev To prevent the same signature from being validated on different accounts owned by the samer signer,
+///      we introduce an anti cross-account-replay layer: the original signed hash (OSH) is wrapped in a
```

Comment:
```suggestion
///      we introduce an anti cross-account-replay layer: the original hash is input into a new EIP-712-compliant hash. The domain separator of this outer hash contains the chain id and address of this contract, so that it cannot be used on two accounts. 
```

Reply:
Done in #3eb8d93.